source/application/utils/file/Path.cpp

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright(c) 2017 - Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#include "../File.h"
+
+std::string Utils::File::GetFileName(const std::string path)
```

Comment:
I mean, `class Path : public std::string` that gives all the functionality of string (and can be passed as a string if we don't add member variables), but with these additional functions to get directory and file name from the class.